synapse/replication/tcp/handler.py

```diff
@@ -71,7 +83,41 @@ def __init__(self, hs):
         self.factory = None  # type: Optional[ReplicationClientFactory]
 
         # The currently connected connections.
-        self.connections = []
+        self.connections = []  # type: List[Any]
```

Comment:
looks like we're relying on these being `BaseReplicationStreamProtocol`. Can you improve the annotation?